src/test/java/io/pravega/connectors/flink/utils/StreamSourceOperatorTestHarness.java

```diff
@@ -52,8 +49,8 @@ public StreamSourceOperatorTestHarness(StreamSource<T, F> operator, int maxParal
         super(operator, maxParallelism, parallelism, subtaskIndex);
         this.sourceOperator = operator;
         this.triggeredCheckpoints = new ConcurrentLinkedQueue<>();
-        this.recordWriters = StreamTask.createRecordWriters(this.config, this.getEnvironment());
-        this.operatorChain = new OperatorChain<>(this.mockTask, recordWriters);
+        this.recordWriterDelegate = StreamTask.createRecordWriterDelegate(this.config, this.getEnvironment());
```

Comment:
Actually we don't need the `recordWriterDelegate` as a member of the class, the below sentence is the only place we need that, we can contruct it below.